File: app/src/main/java/com/github/malitsplus/shizurunotes/data/action/AbnormalStateFieldAction.java
package com.github.malitsplus.shizurunotes.data.action;

import com.github.malitsplus.shizurunotes.R;
import com.github.malitsplus.shizurunotes.common.I18N;
import com.github.malitsplus.shizurunotes.data.Property;

import java.math.RoundingMode;
import java.util.ArrayList;
import java.util.List;

public class AbnormalStateFieldAction extends ActionParameter {

    protected List<ActionValue> durationValues = new ArrayList<>();

    @Override
    protected void childInit() {
        super.childInit();
        durationValues.add(new ActionValue(actionValue1, actionValue2, null));
    }

    @Override
    public String localizedDetail(int level, Property property) {
        return I18N.getString(R.string.Summon_a_field_of_radius_d1_on_s2_to_cast_effect_d3_for_s4_sec,
                (int)actionValue3.value,
                targetParameter.buildTargetClause(),
                getActionNum(actionDetail1),
                buildExpression(level, durationValues, RoundingMode.UNNECESSARY, property));
    }
}

File: app/src/main/java/com/github/nyanfantasia/shizurunotes/data/action/AbnormalStateFieldAction.kt
package com.github.nyanfantasia.shizurunotes.data.action

import com.github.nyanfantasia.shizurunotes.R
import com.github.nyanfantasia.shizurunotes.common.I18N.Companion.getString
import com.github.nyanfantasia.shizurunotes.data.Property
import java.math.RoundingMode
import java.util.*

class AbnormalStateFieldAction : ActionParameter() {
    private var durationValues: MutableList<ActionValue> = ArrayList()
    override fun childInit() {
        super.childInit()
        durationValues.add(ActionValue(actionValue1, actionValue2, null))
    }

    override fun localizedDetail(level: Int, property: Property): String {
        return getString(
            R.string.Summon_a_field_of_radius_d1_on_s2_to_cast_effect_d3_for_s4_sec,
            actionValue3.value.toInt(),
            targetParameter.buildTargetClause(),
            actionDetail1 % 10,
            buildExpression(level, durationValues, RoundingMode.UNNECESSARY, property)
        )
    }
}